packages/formspree-core/src/core.ts

```diff
@@ -75,78 +67,108 @@ export class Client {
     if (this.stripePromise && opts.createPaymentMethod) {
-      // Get Stripe payload
-      const payload = await opts.createPaymentMethod();
-
-      if (payload.error) {
-        // Return the error in case Stripe failed to create a payment method
-        return {
-          response: null,
-          body: {
-            errors: [
-              {
-                code: 'STRIPE_CLIENT_ERROR',
-                message: 'Error creating payment method',
-                field: 'paymentMethod',
-              },
-            ],
-          },
-        };
+      const createPaymentMethodResult = await opts.createPaymentMethod();
+
+      if (createPaymentMethodResult.error) {
+        return new SubmissionErrorResult({
+          code: 'STRIPE_CLIENT_ERROR',
+          field: 'paymentMethod',
+          message: 'Error creating payment method',
+        });
       }
 
       // Add the paymentMethod to the data
-      appendExtraData(data, 'paymentMethod', payload.paymentMethod.id);
+      appendExtraData(
+        data,
+        'paymentMethod',
+        createPaymentMethodResult.paymentMethod.id
+      );
 
       // Send a request to Formspree server to handle the payment method
-      const response = await fetchImpl(url, {
-        ...request,
-        body: serializeBody(data),
-      });
-      const responseData = await response.json();
-
-      // Handle SCA
-      if (
-        responseData &&
-        responseData.stripe &&
-        responseData.stripe.requiresAction &&
-        responseData.resubmitKey
-      ) {
-        return await handleSCA({
-          stripePromise: this.stripePromise,
-          responseData,
-          response,
-          payload,
-          data,
-          fetchImpl,
-          request,
-          url,
-        });
+      const result = await makeFormspreeRequest(data);
+
+      if (result.kind === 'error') {
+        return result;
       }
 
-      return handleLegacyErrorPayload({
-        response,
-        body: responseData,
-      });
-    } else {
-      return fetchImpl(url, request)
-        .then((response) => {
-          return response
-            .json()
-            .then((body: SubmissionBody): SubmissionResponse => {
-              return handleLegacyErrorPayload({ body, response });
-            });
-        })
-        .catch();
+      if (result.kind === 'stripePluginPending') {
+        const stripeResult = await this.stripePromise.handleCardAction(
+          result.paymentIntentClientSecret
+        );
+
+        if (stripeResult.error) {
+          return new SubmissionErrorResult({
+            code: 'STRIPE_CLIENT_ERROR',
+            field: 'paymentMethod',
+            message: 'Stripe SCA error',
+          });
+        }
+
+        // delete 'paymentMethod' to resubmit
+        if (data instanceof FormData) {
+          data.delete('paymentMethod');
+        } else {
+          delete data.paymentMethod;
+        }
```

Comment:
Left the comment because it's unclear if this is intentional in the original implementation (I thought it was a mistake). The original implementation also doesn't handle the `FormData` case.

I looked at the server implementation. The `paymentMethod` must not be passed when performing resubmit (confirmation), so we need to delete it here.